test/Microsoft.Azure.Devices.Edge.Test/PriorityQueues.cs

```diff
@@ -73,105 +148,109 @@ public async Task PriorityQueueModuleToModuleMessages()
                         });
 
-                    Dictionary<string, object> routes = this.BuildRoutes(priorityString.Split(';'), loadGenModuleName, relayerModuleName);
+                    Dictionary<string, object> routes = this.BuildRoutes(testInfo.Priorities, testInfo.Ttls, LoadGenModuleName, RelayerModuleName);
                     builder.GetModule(ModuleName.EdgeHub).WithDesiredProperties(new Dictionary<string, object> { ["routes"] = routes });
                 });
+        }
 
-            EdgeDeployment deployment = await this.runtime.DeployConfigurationAsync(addInitialConfig, token);
-            PriorityQueueTestStatus loadGenTestStatus = await this.PollUntilFinishedAsync(loadGenModuleName, token);
+        private Dictionary<string, object> BuildRoutes(List<int> priorities, List<int> ttls, string sendModule, string receiveModule)
+        {
+            Dictionary<string, object> routes = new Dictionary<string, object>();
+            for (int i = 0; i < priorities.Count; i++)
+            {
+                int ttl = ttls[i];
+                int priority = priorities[i];
+                var routeInfo = new Dictionary<string, object>();
+                routeInfo["route"] = $"FROM /messages/modules/{sendModule}/outputs/pri{priority} INTO BrokeredEndpoint('/modules/{receiveModule}/inputs/input1')";
 
-            Action<EdgeConfigBuilder> addRelayerConfig = new Action<EdgeConfigBuilder>(
-                builder =>
+                // If we encounter "Default" in the priority list, don't add a priority - the default priority will automatically get picked up
+                if (priority >= 0)
                 {
-                    builder.AddModule(relayerModuleName, relayerImage)
-                        .WithEnvironment(new[]
-                        {
-                            ("receiveOnly", "true"),
-                            ("uniqueResultsExpected", loadGenTestStatus.ResultCount.ToString())
-                        });
-                });
+                    routeInfo["priority"] = priority;
+                }
 
-            deployment = await this.runtime.DeployConfigurationAsync(addInitialConfig + addRelayerConfig, token, false);
-            await this.PollUntilFinishedAsync(relayerModuleName, token);
+                if (ttl >= 0)
+                {
+                    routeInfo["timeToLiveSecs"] = ttl;
+                }
 
-            HttpClient client = new HttpClient();
-            HttpResponseMessage response = await client.GetAsync("http://localhost:5001/api/report");
-            var jsonstring = await response.Content.ReadAsStringAsync();
-            bool isPassed = (bool)JArray.Parse(jsonstring)[0]["IsPassed"];
-            if (!isPassed)
+                routes.Add($"LoadGenToRelayer{priority}", routeInfo);
+            }
+
+            return routes;
+        }
+
+        private List<int> BuildDefaultTtls(int numberOfTTLs)
+        {
+            var ttls = new List<int>();
```

Comment:
return new List<int>(Enumerable.Repeat(0, numberOfTTLs));

Reply:
I think you can combine BuildDefaultTtls and BuildTtls method using Option<int> ttlThreshold.